## CI failure diagnosis: `build-pre-release` - upstream package publishing gap (infrastructure, not code)

The `build-pre-release` check (run [32279383448](https://github.com/credfeto/credfeto-changelog-manager/actions/runs/32279383448/job/96154272652)) is failing during the "Install Build tools" step, not during our code's build/test:

```
Version 474.2.10.2548 of package funfair.buildcheck.linux-x64 is not found in NuGet feeds ***.
##[error]Process completed with exit code 1.
```

Root cause: `.github/actions/dotnet-build-check/action.yml` installs `FunFair.BuildCheck` with `tool-version: "latest"`. NuGet currently resolves `latest` for the meta-package `funfair.buildcheck` to `474.2.10.2548`, but the RID-specific companion package `funfair.buildcheck.linux-x64` that the tool install needs has not been published past `474.2.10.2518-dotnet-sdk-10-0`. I verified directly against the NuGet flat-container index for both packages - confirmed `474.2.10.2548` exists for `funfair.buildcheck` but is absent from `funfair.buildcheck.linux-x64`'s version list.

This is an upstream release-pipeline gap in `FunFair.BuildCheck` itself (the RID-specific package lagging the meta-package), not something caused by or fixable in this PR's diff - `tool-version: "latest"` means every branch/PR in this repo would hit the identical failure right now, including `main`. No workaround was applied (e.g. pinning an older tool version), since `dotnet-build-check/action.yml` is maintained upstream in `funfair-server-template`, not in this repo.

Note: `IMAGE_SHA_DEVELOPMENT_AGENT` was not obtainable in this session (blocked from reading it), so the usual auto-clear marker is omitted below. This block will **not** auto-clear when a differently-built execution image appears - please re-run `build-pre-release` manually once the upstream `funfair.buildcheck.linux-x64` package catches up (or unblock manually).

<!-- orchestrator:env-block image-sha=unavailable -->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ Please ADD ALL Changes to the UNRELEASED SECTION and not a specific release
 - Dependencies - Updated Meziantou.Analyzer to 3.0.156
 - Dependencies - Updated xunit.analyzers to 2.0.0
 - Dependencies - Updated xunit.v3 to 4.0.0
+- TBD - to be finalized after review (issue #336)
 ### Deprecated
 ### Removed
 ### Deployment Changes

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,7 +21,7 @@ Please ADD ALL Changes to the UNRELEASED SECTION and not a specific release
 - Dependencies - Updated Meziantou.Analyzer to 3.0.156
 - Dependencies - Updated xunit.analyzers to 2.0.0
 - Dependencies - Updated xunit.v3 to 4.0.0
-- TBD - to be finalized after review (issue #336)
+- ChangeLogUpdater - Hoisted the loop-invariant requested-version parse out of ValidateVersionNotExists so it is parsed once per --create-release invocation instead of once per existing release
 ### Deprecated
 ### Removed
 ### Deployment Changes

--- a/COVERAGE.md
+++ b/COVERAGE.md
@@ -24,4 +24,4 @@ excluded
 
 ---
 
-Captured at commit `e8137c9e842da6b9eefb3119aaec0c4b292f9f10` on 2026-08-18.
+Captured at commit `f5b7229ca2cd1eb2599a57999f710a4cd25fa954` on 2026-08-19.

--- a/src/Credfeto.ChangeLog/Services/ChangeLogUpdater.cs
+++ b/src/Credfeto.ChangeLog/Services/ChangeLogUpdater.cs
@@ -252,6 +252,8 @@ public sealed class ChangeLogUpdater : IChangeLogUpdater
     )]
     private static void ValidateVersionNotExists(in ImmutableArray<ChangeLogRelease> releases, string version)
     {
+        bool requestedParsed = Version.TryParse(version, out Version? requested);
+
         foreach (ChangeLogRelease release in releases)
         {
             if (release.Version.EqualsOrdinal(version))
@@ -259,11 +261,7 @@ public sealed class ChangeLogUpdater : IChangeLogUpdater
                 throw new ReleaseAlreadyExistsException($"Release {version} already exists");
             }
 
-            if (
-                Version.TryParse(release.Version, out Version? existing)
-                && Version.TryParse(version, out Version? requested)
-                && existing > requested
-            )
+            if (requestedParsed && Version.TryParse(release.Version, out Version? existing) && existing > requested)
             {
                 throw new ReleaseTooOldException(
                     $"Release {release.Version} already exists and is newer than {version}"

